### DISCUSSION TITLE
Wiki.js with Nginx Basic Authentication?

### DISCUSSION THREAD

**Discussion by alvinj** [requarks/wiki#4148] [Help / Questions] [answered]
Hi, I just installed Wiki.js (for the first time) behind an Nginx reverse-proxy on an Ubuntu 20.04 system. Everything was working fine, and then when I added a “Basic Authentication” configuration to Nginx, I started seeing the “Network error: JSON. parse” error, as shown in the attached image.

![network-error](https://user-images.githubusercontent.com/309803/123888530-ecc57600-d910-11eb-840b-776dee74bd93.jpg)

I searched these discussions and found that this issue was already reported [here](https://github.com/Requarks/wiki/discussions/3244), but no solution was shown.

So I’m wondering, is there any way to get Wiki.js to work with Nginx basic authentication (i.e., with `auth_basic` and `auth_basic_user_file` settings in the `nginx.conf` file)? For now there’s no need for anyone other than me to be able to access the Wiki.js login page.

Or, is there a better way to secure this? At the moment this installation is just for me, but I do want to secure it. (FWIW, I have also limited access to it by IP range in `nginx.conf`.)

For the record, these are the versions I’m using:

```
OS: Ubuntu 20.04
Wiki.js version: 2.5.201
Database engine: PostgreSQL 12+214
```

Thanks!

**Comment by rwky** [comment, marked as answer]
The error is because the /graphql route sends a bearer token which is incompatible with Nginx basic auth. If you disable basic auth for that path but allow it for other pages it'll prevent public access to the wiki but still allow admins to do things (I assume the graphql route doesn't have any public methods I've not checked).

Example config:

```
    location /graphql {
        auth_basic off;
        proxy_pass http://127.0.0.1:3000;
    }

    location / {
        auth_basic "Please log in:";
        auth_basic_user_file /path/to/file;
        proxy_pass http://127.0.0.1:3000;
    }

```

**Reply by alvinj** [reply]
Thank you for your reply, I appreciate it. I stopped using Wiki.js because of this issue, but if/when I try it again, I’ll give this a shot. 

Please feel free to close this bug as I won’t be able to test it any time soon. Thanks again!
